docs/faq.md

```diff
@@ -0,0 +1,122 @@
+
+## How do I install my favourite package?
+
+You should use the `emacsPackagesOverlay` attribute. Here's an example that installs `magit-delta` which depends on Git:
+
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    magit-delta = super.magit-delta.overrideAttrs (esuper: {
+      buildInputs = esuper.buildInputs ++ [ pkgs.git ];
+    });
+  }
+};
+```
+
+### But what if my package is from a Git source, and isn't on ELPA?
+
+This requires a few more lines, but it isn't by any means difficult. For an example, this installs `idris2-mode` which isn't on ELPA, but is hosted on GitHub:
+
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    idris2-mode = self.trivialBuild {
+      pname = "idris2-mode";
+      ename = "idris2-mode";
+      version = "0.0.0";
+      buildInputs = [ self.prop-menu ];
+      src = pkgs.fetchFromGitHub {
+        owner = "idris-community";
+        repo = "idris2-mode";
+        rev = "4a3f9cdb1a155da59824e39f0ac78ccf72f2ca97";
+        sha256 = "sha256-TxsGaG2fBRWWP9aas59kiNnUVD4ZdNlwwaFbM4+n81c=";
+      };
+    };
+  }
+};
+```
+
+### Help! My favourite package doesn't work after adding it in!
+
+This usually happens when the package depends on either some ELPA package or a normal package that you need to add into the `buildInputs` to make it work. 
+Let's take the above `idris2-mode` package, and I'll remove the `buildInputs` line:
+
+**WARNING**: THIS IS ERRONEOUS CODE FOR DEMONSTRATION. DO NOT USE THIS SNIPPET, THE ABOVE ONE IS THE FUNCTIONING VERSION.
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    idris2-mode = self.trivialBuild {
+      pname = "idris2-mode";
+      ename = "idris2-mode";
+      version = "0.0.0";
+      src = pkgs.fetchFromGitHub {
+        owner = "idris-community";
+        repo = "idris2-mode";
+        rev = "4a3f9cdb1a155da59824e39f0ac78ccf72f2ca97";
+        sha256 = "sha256-TxsGaG2fBRWWP9aas59kiNnUVD4ZdNlwwaFbM4+n81c=";
+      };
+    };
+  }
+};
+```
+
+This will error with (this is a part of the build log):
+
+```
+In toplevel form:
+idris2-hole-list.el:27:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In toplevel form:
+idris2-info.el:29:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In idris2-ipkg-pkgs-flags-for-current-buffer:
+idris2-ipkg-mode.el:372:2: Warning: docstring wider than 80 characters
+
+In toplevel form:
+idris2-mode.el:20:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In idris2-prover-end:
+idris2-prover.el:378:2: Warning: docstring wider than 80 characters
+
+In toplevel form:
+idris2-repl.el:29:2: Error: Cannot open load file: No such file or directory, prop-menu
+```
+
+The way to fix it is by just adding the dependency it's complaining about.
```

Comment:
Not sure if this section is really useful. This package did complain very loudly about its missing dependency, but this is not always true. Also in general, I think debugging dependency builds is like debugging issues with any other Nix derivation, that I think we should assume the user from NDE already have.

Reply:
If you do want to keep this section here, I think it is better to explain the basics of debugging in Nix. For example, showing the `nix log` command to get the full logs of a failed build.

Reply:
I think having the basics of debugging in nix would be perfect for this section. It won't really be a "nix tutorial", but just a couple references to external pages that might explain how debugging in nix works, or just listing a few helpful commands to read the documentation. Thanks.